lib/inherited_resources/url_helpers.rb

```diff
@@ -216,16 +216,20 @@ def generate_url_and_path_helpers(prefix, name, resource_segments, resource_ivar
       prefix = prefix ? "#{prefix}_" : ''
       ivars << (ivars.empty? ? 'given_options' : ', given_options')
 
+      given_args_transform = defined?(ActionController::Parameters) ? 'given_args = given_args.collect { |arg| arg.respond_to?(:permitted?) ? arg.to_h : arg }' : ''
```

Comment:
You don't need to check if `ActionController::Parameters` is defined anymore with the actual code.